src/index.js

```diff
@@ -151,31 +151,25 @@ function check(autoCheckElement: AutoCheckElement) {
       if (response.status !== 200) {
         throw new ErrorWithResponse(response.statusText, response)
       }
-      return response.text()
+      return response
     })
-    .then(message => {
+    .then(response => {
       autoCheckElement.dispatchEvent(new CustomEvent('load'))
       if (autoCheckElement.required) {
         input.setCustomValidity('')
       }
-      input.dispatchEvent(new CustomEvent('auto-check-success', {detail: {message}, bubbles: true}))
+      input.dispatchEvent(new CustomEvent('auto-check-success', {detail: {response: response.clone()}, bubbles: true}))
 
       // Mark the component as not being in-flight any more.
       abortControllers.delete(autoCheckElement)
     })
-    .catch(async error => {
-      const message = await error.response.text()
-      const contentType = error.response.headers.get('Content-Type')
-
+    .catch(error => {
       if (autoCheckElement.required) {
         input.setCustomValidity('Input is not valid')
       }
       autoCheckElement.dispatchEvent(new CustomEvent('error'))
```

Comment:
The `error` event should be dispatched from the catch block surrounding the `fetch` call because it signals only a network failure. The `auto-check-error` event should be dispatched only when we receive a response from the server that indicates the check validation failed.

Reply:
Something like what I've pushed?